subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationIdentifierRegistry.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.operations;
+
+public class BuildOperationIdentifierRegistry {
+    private static ThreadLocal<OperationIdentifier> localOperationId = new ThreadLocal<OperationIdentifier>();
+
+    public static OperationIdentifier getCurrentOperationIdentifier() {
+        return localOperationId.get();
+    }
+
+    public static void setCurrentOperationIdentifier(OperationIdentifier operationId) {
+        if (operationId == null) {
```

Comment:
It might be clearer if we'd introduce another method for removing the operation Id. Maybe something like `clearCurrentOperationIdentifier()`.